wpilib/wpilib/nidecbrushless.py

```diff
@@ -0,0 +1,124 @@
+from .motorsafety import MotorSafety
+from .interfaces.speedcontroller import SpeedController
+
+
+class NidecBrushless(LiveWindowSendable, MotorSafety, SpeedController):
+    def __init__(self, pwmChannel, dioChannel):
+        """
+        :param pwmChannel: The PWM channel that the Nidec Brushless controller is attached to.
+                0-9 are on-board, 10-19 are on the MXP port
+        :param dioChannel: The DIO channel that the Nidec Brushless controller is attached to.
+                0-9 are on-board, 10-25 are on the MXP port
+        """
+        self.dio = DigitalOutput(dioChannel)
+        self.dio.setPWMRate(15625)
+        self.dio.enablePWM(0.5)
+
+        self.pwm = PWM(pwmChannel)
+        self.pwm.setRaw(0xffff)
+
+        self.speed = 0.0
+        self.isInverted = False
+        self.valueEntry = None
+        self.valueListener = None
+
+        LiveWindow.addActuator("Nidec Brushless", pwmChannel, self)
+        hal.report(hal.UsageReporting.kResourceType_NidecBrushless, pwmChannel)
+
+    def set(self, speed):
+        """ 
+        Set the PWM value.
+    
+        The PWM value is set using a range of -1.0 to 1.0, appropriately scaling the value for the FPGA.
+
+       :param speed: The speed value between -1.0 and 1.0 to set.
+        """
+        self.speed = speed
+        self.dio.updateDutyCycle(0.5 + 0.5 * (-speed if self.isInverted else speed))
+        self.feed()
+
+    def get(self):
+        """
+        Get the recently set value of the PWM.  
+
+        :returns: The most recently set value for the PWM between -1.0 and 1.0. 
+        :rtype: float
+        """
+        return self.speed
+
+    def setInverted(self, isInverted):
+        """
+        :type isInverted: bool
+        """
+        self.isInverted = isInverted
+
+    def getInverted(self):
+        """
+        :rtype: bool
+        """
+        return self.isInverted
+
+    def pidWrite(self, output):
+        """ 
+        Write out the PID value as seen in the PIDOutput base object. 
+
+        :param output: Write out the PWM value as was found in the PIDController
+        :type output: float
+        """
+        self.set(output)
+
+    def stopMotor(self):
+        """
+        Stop the motor. This is called by the MotorSafetyHelper object
+        when it has a timeout for this PWM and needs to stop it from running.
+        """
+        self.disable()
+
+    def getDescription(self):
+        """
+        :rtype: str
+        """
+        return "Nidec " + self.getChannel()
+
+    def disable(self):
+        self.dio.updateDutyCycle(0.5)
+
+    def getChannel(self):
+        """
+        Gets the channel number associated with the object.
+        :returns: The channel number.
+        :rtype: int
+        """
+        return self.pwm.getChannel()
+
+    def getSmartDashboardType(self):
+        """
+        Live Window code, only does anything if live window is activated.
+        """
```

Comment:
Upstream does not have a corresponding Javadoc for this method.

Reply:
I'm ok with this.

Reply:
Yeah, but this is a comment in upstream, which is intended to refer to the several methods below as well.